src/main/java/com/atlassian/db/replica/internal/aurora/AuroraClusterDiscovery.java

```diff
@@ -6,22 +6,34 @@
 import java.sql.Connection;
 import java.sql.SQLException;
 import java.util.Collection;
+import java.util.Properties;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 import java.util.function.Supplier;
 
 import static java.util.stream.Collectors.toList;
 
 public final class AuroraClusterDiscovery {
     private final AuroraConnectionDetails auroraConnectionDetails;
+    private final BiFunction<String, Properties, AuroraConnectionSupplier> connectionPerReplicaProvider;
 
-    public AuroraClusterDiscovery(AuroraConnectionDetails auroraConnectionDetails) {
+    public AuroraClusterDiscovery(
+        AuroraConnectionDetails auroraConnectionDetails,
```

Comment:
We should not need AuroraConnectionDetails anymore.

Reply:
removed